poet/poet.py

```diff
@@ -91,10 +93,12 @@ class PipSourceMetadataException(Exception):
 @dataclass
 class PackageMetadata:
     name: str
-    homepage: str
-    url: str
-    checksum: str
+    homepage: str = field(optional=True)
+    download_url: Optional[str] = None
+    checksum: Optional[str] = None
     checksum_type: str = field(default="sha256", init=False)
+    source_file: Optional[Path] = None
+    version: Optional[Version] = None
```

Comment:
here's a way to filter `version` from a returned dict?
```suggestion
    version: Optional[Version] = None
    
    def asdict(self):
        exclude_keys = ["version"]
        return dict(filter(lambda x: x[0] not in exclude_keys, super().asdict()))
```
the idea here is that `research_package()` would return `package_metadata.asdict()`